swarm/storage/resource_test.go

```diff
@@ -456,3 +458,99 @@ func (self *testValidator) checkAccess(name string, address common.Address) (boo
 func (self *testValidator) nameHash(name string) common.Hash {
 	return self.hashFunc(name)
 }
+
+func TestResourceMultihash(t *testing.T) {
+
+	// signer containing private key
+	//	signer, err := newTestSigner()
+	//	if err != nil {
+	//		t.Fatal(err)
+	//	}
+
+	// make fake backend, set up rpc and create resourcehandler
+	backend := &fakeBackend{
+		blocknumber: int64(startBlock),
+	}
+
+	// set up rpc and create resourcehandler
+	rh, _, _, teardownTest, err := setupTest(backend, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer teardownTest()
+
+	// create a new resource
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
This context usage is ineffective. Since `cancel` is being deferred, it will only be called _after_ the call to `NewResource` returns, therefore nothing will get cancelled.

We should just pass `context.Background()` directly to `NewResource`.